src/unix/core.c

```diff
@@ -1226,3 +1227,26 @@ void uv_os_free_passwd(uv_passwd_t* pwd) {
 int uv_os_get_passwd(uv_passwd_t* pwd) {
   return uv__getpwuid_r(pwd);
 }
+
+/* Leverage RFC1323: TCP extensions for high performance
+ * https://tools.ietf.org/html/rfc1323
+ * This could be as simple as getsockopt(socket) to get the buffer
+ * sizes, and then applying to the stream's chunk read size. But
+ * given that the behavior is platform dependant, we go with the 
+ * OS configuration to get the values. Moreover in Linux, we want to
+ * leverage the maximum size not the default, which are different.
+ */
+
+int uv_tcp_enable_largesocket(uv_tcp_t *tcp) {
+#if defined(_AIX)                 ||                                      \
+    defined(__linux__)            ||                                      \
+    defined(__APPLE__)            ||                                      \
+    defined(__OpenBSD__)          ||                                      \
+    defined(__FreeBSD__)          ||                                      \
+    defined(__NetBSD__)
+  return uv__tcp_enable_largesocket(tcp);
+#else
+  /* Implement me. */
+  return UV_ENOSYS;
```

Comment:
I'm not sure its better, but the other options is for there be an implementation of `uv__tcp_enable_largesocket()` for every platform, where some simply return `UV_ENOSYS`? That would allow the platform list to not have to be maintained in this source file, and the list is present in a couple places, if that's necessary, perhaps there should be a single platform list in uv-common.h that is used to define a `UV__HAS_LARGESSOCKET`, and the ifdefs in the .c files would use that, so its less likely that the platform lists get out of sync.